### ISSUE TITLE
azurerm_linux_virtual_machine disable_password_authentication=false not working

### ISSUE THREAD

**Issue by shanep2300** [hashicorp/terraform-provider-azurerm#18069]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.0.10

### AzureRM Provider Version

3.19.1

### Affected Resource(s)/Data Source(s)

azurerm_linux_virtual_machine

### Terraform Configuration Files

```hcl
provider "azurerm" {
  subscription_id = "${var.management_subscription_id}"
  alias = "subscription_mgmt"  
  features {}
}

locals {
  vm_os_disk_name = "os-disk"
  resource_group_name = "${var.mandatory_prefix}-${var.resource_group_name}"
  vnet_resource_group_name = "${var.mandatory_prefix}-${var.vnet_resource_group_name}"  
  local_lb_ports = "${var.lb_port}"
}

data "azurerm_subnet" "primary_subnet" {
  name = "${var.nic_subnet_name}"
  virtual_network_name = "${var.mandatory_prefix}-${var.nic_vnet_name}"
  resource_group_name ="${local.vnet_resource_group_name}"
}

data "azurerm_shared_image" "share_custom_image" {
  provider = "azurerm.subscription_mgmt"
  name = "${var.shared_image_name}" 
  gallery_name = "${var.shared_gallery_name}"
  resource_group_name = "${var.shared_gallery_resource_group}"
}

resource "azurerm_network_interface" "generic_nic" {
  count = "${var.number_of_vms}"
  
  name = "${var.mandatory_prefix}-${var.host_name}-${var.nic_name}-${count.index + 1}"
  location = "${var.resource_location}"
  resource_group_name = "${local.resource_group_name}"

  ip_configuration {
    name = "${var.mandatory_prefix}-${var.nic_name}-cfg-${count.index + 1}"
    subnet_id = "${data.azurerm_subnet.primary_subnet.id}"
    private_ip_address_allocation = "${var.nic_address_allocation}"
    private_ip_address = "${var.nic_ip_address}"
    primary = "${var.is_primary_nic}"
  }

  tags = "${var.mandatory_tags}"
}

resource "azurerm_availability_set" "availability_set" {  
  name = "${var.host_name}-${var.availability_set}"
  location = "${var.resource_location}"
  resource_group_name = "${local.resource_group_name}"
  
  tags = "${var.mandatory_tags}"
}

resource "azurerm_linux_virtual_machine" "generic_linux_vm" {
  count = "${var.number_of_vms}"  
  
  name = "${var.host_name}"
  admin_username = "${var.admin_username}"
  admin_password = "${var.admin_password}"
  disable_password_authentication = false
  location = "${var.resource_location}"
  resource_group_name = "${local.resource_group_name}" 
  availability_set_id = "${azurerm_availability_set.availability_set.id}"
  network_interface_ids = ["${azurerm_network_interface.generic_nic["${count.index}"].id}"]
  size = "${var.machine_size}" 
  source_image_id = "${data.azurerm_shared_image.share_custom_image.id}"

  os_disk {
    name = "${var.mandatory_prefix}-${var.host_name}-vm-${count.index + 1}-${local.vm_os_disk_name}"
    caching = "ReadWrite"
    disk_size_gb = "${var.vm_os_disk_size}"
    storage_account_type = "${var.managedos_disk_type}" 
  }

  tags = "${var.mandatory_tags}"

  depends_on = [azurerm_network_interface.generic_nic,azurerm_availability_set.availability_set]
}

resource "azurerm_lb" "lb" {
  name                = "${var.mandatory_prefix}-${var.host_name}-${var.lb_name}"
  location            = "${var.resource_location}"
  resource_group_name = "${local.resource_group_name}"  
  sku = "Standard"

  frontend_ip_configuration {
    name   = "${var.lb_frontendip_name}"   
    subnet_id = "${data.azurerm_subnet.primary_subnet.id}"
  }

  tags = "${var.mandatory_tags}"

  depends_on = [azurerm_linux_virtual_machine.generic_linux_vm]
}

resource "azurerm_lb_backend_address_pool" "lb_backend" { 
  name                = "${var.mandatory_prefix}-${var.lb_backend_name}-backend-pool"
  loadbalancer_id     = "${azurerm_lb.lb.id}"

  depends_on = [azurerm_lb.lb]
}

resource "azurerm_network_interface_backend_address_pool_association" "vm_association" {
  count = "${var.number_of_vms}"

  network_interface_id    = "${azurerm_network_interface.generic_nic["${count.index}"].id}"
  ip_configuration_name   = "${var.mandatory_prefix}-${var.nic_name}-cfg-${count.index + 1}"
  backend_address_pool_id = "${azurerm_lb_backend_address_pool.lb_backend.id}"

  depends_on = [azurerm_lb_backend_address_pool.lb_backend]
}

resource "azurerm_lb_probe" "lb_probe" {
  count = "${length(local.local_lb_ports)}"

  loadbalancer_id     = "${azurerm_lb.lb.id}"
  name                = "${local.local_lb_ports[count.index].name}"
  protocol            = "${local.local_lb_ports[count.index].protocol}"
  port                = "${local.local_lb_ports[count.index].port}"
  interval_in_seconds = "300"
  number_of_probes    = "${var.lb_probe_unhealthy_threshold}"

  depends_on = [azurerm_network_interface_backend_address_pool_association.vm_association]
}

resource "azurerm_lb_rule" "lb_rule" {  
  count = "${length(local.local_lb_ports)}"

  loadbalancer_id                = "${azurerm_lb.lb.id}"
  name                           = "${local.local_lb_ports[count.index].name}"
  protocol                       = "${local.local_lb_ports[count.index].protocol}"
  backend_port                   = "${local.local_lb_ports[count.index].port}"
  frontend_port                  = "${local.local_lb_ports[count.index].frontendport}"
  frontend_ip_configuration_name = "${var.lb_frontendip_name}"
  enable_floating_ip             = false
  backend_address_pool_ids       = ["${azurerm_lb_backend_address_pool.lb_backend.id}"]
  idle_timeout_in_minutes        = 5
  probe_id                       = "${element(azurerm_lb_probe.lb_probe.*.id,count.index)}"

  depends_on = [azurerm_lb_probe.lb_probe]
}
```

### Debug Output/Panic Output

```shell
N/A
```

### Expected Behaviour

SSH into machine with admin and password

### Actual Behaviour

Incorrect user/password

### Steps to Reproduce

Apply.
Attempt to SSH to machine.
Can't connect.

Temporary fix: 
Go to Azure portal, find VM, click Reset Password, type in same user and same password from variables.tf.
Now, I can connect.

**Comment by myc2h6o** [comment]
Hi @shanep2300 thanks for opening the issue! The config shared in the issue is not the complete so I tried with below config trying to repro the issue. It is working fine in my environment without the need of resetting password on Azure Portal. Could you check if this works in your environment and see if there is any difference with creating the image/vm?

```
provider "azurerm" {
  features {}
}

provider "null" {
}

variable "prefix" {
  type    = string
  default = "test"
}

# source virtual machine to create image version
resource "azurerm_resource_group" "source" {
  name     = "${var.prefix}-source-rg"
  location = "eastus"
}

resource "azurerm_virtual_network" "source" {
  name                = "${var.prefix}-source-vnet"
  address_space       = ["10.0.0.0/16"]
  location            = azurerm_resource_group.source.location
  resource_group_name = azurerm_resource_group.source.name
}

resource "azurerm_subnet" "source" {
  name                 = "subnet"
  resource_group_name  = azurerm_resource_group.source.name
  virtual_network_name = azurerm_virtual_network.source.name
  address_prefixes     = ["10.0.2.0/24"]
}

resource "azurerm_network_interface" "source" {
  name                = "${var.prefix}-source-nic"
  location            = azurerm_resource_group.source.location
  resource_group_name = azurerm_resource_group.source.name

  ip_configuration {
    name                          = "internal"
    subnet_id                     = azurerm_subnet.source.id
    private_ip_address_allocation = "Dynamic"
  }
}

resource "azurerm_linux_virtual_machine" "source" {
  name                = "${var.prefix}-source-vm"
  resource_group_name = azurerm_resource_group.source.name
  location            = azurerm_resource_group.source.location
  size                = "Standard_D2s_v3"

  disable_password_authentication = false
  admin_username                  = "adminuser"
  admin_password                  = "P@$$w0rd1234!"

  network_interface_ids = [
    azurerm_network_interface.source.id,
  ]
  os_disk {
    caching              = "ReadWrite"
    storage_account_type = "Standard_LRS"
  }
  source_image_reference {
    publisher = "Canonical"
    offer     = "UbuntuServer"
    sku       = "16.04-LTS"
    version   = "latest"
  }
}

resource "null_resource" "deallocate_vm" {
  triggers = {
    vm_id = azurerm_linux_virtual_machine.source.id
  }

  provisioner "local-exec" {
    command = "az vm deallocate --resource-group ${azurerm_resource_group.source.name} --name ${azurerm_linux_virtual_machine.source.name}"
  }
}

resource "null_resource" "generalize_vm" {
  triggers = {
    vm_id = azurerm_linux_virtual_machine.source.id
  }

  provisioner "local-exec" {
    command = "az vm generalize --resource-group ${azurerm_resource_group.source.name} --name ${azurerm_linux_virtual_machine.source.name}"
  }

  depends_on = [
    null_resource.deallocate_vm
  ]
}

resource "azurerm_image" "source" {
  name                      = "${var.prefix}-source-image"
  location                  = azurerm_resource_group.source.location
  resource_group_name       = azurerm_resource_group.source.name
  source_virtual_machine_id = azurerm_linux_virtual_machine.source.id

  depends_on = [
    null_resource.generalize_vm
  ]
}

resource "azurerm_resource_group" "target" {
  name     = "${var.prefix}-target-rg"
  location = "eastus"
}

resource "azurerm_shared_image_gallery" "target" {
  name                = "${var.prefix}targetgallery"
  resource_group_name = azurerm_resource_group.target.name
  location            = azurerm_resource_group.target.location
}

resource "azurerm_shared_image" "target" {
  name                = "${var.prefix}-target-image"
  gallery_name        = azurerm_shared_image_gallery.target.name
  resource_group_name = azurerm_resource_group.target.name
  location            = azurerm_resource_group.target.location
  os_type             = "Linux"
  specialized         = false
  identifier {
    publisher = "testpublisher"
    offer     = "testoffer"
    sku       = "testsku"
  }
}

resource "azurerm_shared_image_version" "target" {
  name                = "0.0.1"
  gallery_name        = azurerm_shared_image_gallery.target.name
  image_name          = azurerm_shared_image.target.name
  resource_group_name = azurerm_resource_group.target.name
  location            = azurerm_resource_group.target.location
  managed_image_id    = azurerm_image.source.id

  target_region {
    name                   = azurerm_resource_group.target.location
    regional_replica_count = 1
  }
}

resource "azurerm_virtual_network" "target" {
  name                = "${var.prefix}-target-vnet"
  address_space       = ["10.0.0.0/16"]
  location            = azurerm_resource_group.target.location
  resource_group_name = azurerm_resource_group.target.name
}

resource "azurerm_subnet" "target" {
  name                 = "subnet"
  resource_group_name  = azurerm_resource_group.target.name
  virtual_network_name = azurerm_virtual_network.target.name
  address_prefixes     = ["10.0.2.0/24"]
}

resource "azurerm_public_ip" "target" {
  name                = "${var.prefix}-target-ip"
  resource_group_name = azurerm_resource_group.target.name
  location            = azurerm_resource_group.target.location
  allocation_method   = "Static"
}

resource "azurerm_network_interface" "target" {
  name                = "${var.prefix}-target-nic"
  location            = azurerm_resource_group.target.location
  resource_group_name = azurerm_resource_group.target.name

  ip_configuration {
    name                          = "internal"
    subnet_id                     = azurerm_subnet.target.id
    private_ip_address_allocation = "Dynamic"
    public_ip_address_id          = azurerm_public_ip.target.id
  }
}


resource "azurerm_network_security_group" "target" {
  name                = "${var.prefix}-nsg"
  location            = azurerm_resource_group.target.location
  resource_group_name = azurerm_resource_group.target.name

  security_rule {
    name                       = "ssh"
    priority                   = 100
    direction                  = "Inbound"
    access                     = "Allow"
    protocol                   = "Tcp"
    source_port_range          = "*"
    destination_port_range     = "22"
    source_address_prefix      = "*"
    destination_address_prefix = "*"
  }
}

resource "azurerm_subnet_network_security_group_association" "target" {
  subnet_id                 = azurerm_subnet.target.id
  network_security_group_id = azurerm_network_security_group.target.id
}

resource "azurerm_availability_set" "availability_set" {
  name                = "${var.prefix}-aset"
  location            = azurerm_resource_group.target.location
  resource_group_name = azurerm_resource_group.target.name
}

resource "azurerm_linux_virtual_machine" "target" {
  count = "1"

  name                            = "${var.prefix}-target-vm"
  admin_username                  = "adminuser2"
  admin_password                  = "P@$$w0rd1234!2"
  disable_password_authentication = false
  location                        = azurerm_resource_group.target.location
  resource_group_name             = azurerm_resource_group.target.name
  availability_set_id             = azurerm_availability_set.availability_set.id
  network_interface_ids           = ["${azurerm_network_interface.target.id}"]
  size                            = "Standard_D2s_v3"
  source_image_id                 = azurerm_shared_image_version.target.id

  os_disk {
    name                 = "${var.prefix}-disk"
    caching              = "ReadWrite"
    storage_account_type = "Standard_LRS"
  }
}

output "ssh_command" {
  value = "ssh adminuser2@${azurerm_public_ip.target.ip_address}" // with password P@$$w0rd1234!2
}
```

**Comment by shanep2300** [comment]
Hi @myc2h6o, I have updated the op with the full tf file for creating our VMs. As for your `azurerm_linux_virtual_machine` resource, I believe everything we're doing is the exact same. I'll create a quick test VM and will report back.

**Comment by shanep2300** [comment]
I'm nervous that it might be something very simple I'm overlooking... Password still doesn't get added to linux for some reason.

**Comment by pativadar** [comment]
anyluck ?

**Comment by shanep2300** [comment]
Well, I got it working.

Still don't understand why, but the image was being created with the user set as centos. When I went into the portal, after creating the VM, the ssh cmd had the username from the tfvars file, which means it was picking up the values. I changed the image creation user to be the same as VM user and it worked.

Doubt this will help anyone that pumps into this issue, but it finally works now!
